### ISSUE TITLE
auro-popover: Popover content placement inside auro drawer

### ISSUE THREAD

**Issue by tfales** [AlaskaAirlines/auro-popover#41]
## Describe the bug
When using auro-popover with an auro-icon trigger _inside_ of an **auro-drawer**, the content is not centered on the trigger element. See screenshot for example. 

## Code Sample
```
<auro-popover>
   <div class="content">
    <div><strong>Premium Class benefits</strong></div>
    <div>
      <ul>
        <li class="text"><strong>More legroom.</strong> 4 inches more than standard Main Cabin seats.</li>
        <li class="text"><strong>Priority boarding.</strong> Board early, access overhead bin space, and be among the
          first to deplane.</li>
        <li class="text"><strong>Drinks.</strong> Enjoy free beer, wine, and cocktails on flights over 350 miles.
          Seat assignments and actual availability are subject to change.</li>
      </ul>
    </div>
  </div>
  <auro-icon class="icon"
    slot="trigger"
    category="alert"
    name="information-stroke"></auro-icon>
</auro-popover>
```

## Expected behavior
When hovering over the trigger element (icon), the popover content is centered above or below the trigger element (icon). 

## Screenshots
**Note:** This is inside of an auro-drawer.
![image](https://github.com/AlaskaAirlines/auro-popover/assets/86446469/5dd6753a-8950-435a-a8bb-69bc7a9246f1)

## Desktop (please complete the following information):
 - OS: Windows
 - Browser: Chrome
 - Version: Chrome 113.0.5672.126, Auro Popover 2.2.0

## Additional context
Email: trenton.fales@alaskaair.com - feel free to follow up with any questions.

**Comment by blackfalcon** [comment]
The popover effect is managed via popover.js and the vast majority of the UI code is written from a relative perspective of there the trigger is. 

What is unknown in this scenario is your code and how that may effect the positioning of the element. 

This issue is not appearing on the AuroDocsite.

**Comment by blackfalcon** [comment]
I was able to successfully reproduce this issue within the drawer demo. This is even worse when used with a dialog. See screen shots below. 

**Dialog**
<img width="643" alt="Screen Shot 2023-05-19 at 1 01 11 PM" src="https://github.com/AlaskaAirlines/auro-popover/assets/181089/c4ee1660-0926-47e7-a20a-419e5785a0c2">

**Drawer**
<img width="486" alt="Screen Shot 2023-05-19 at 1 01 02 PM" src="https://github.com/AlaskaAirlines/auro-popover/assets/181089/1e5abf86-8952-4a5d-aeaa-b162113102a1">

**Comment by Patrick-Daly-AA** [comment]
moving to ready to work for investigation by Auro team in the near future.

**Comment by fajar-apri-alaska** [comment]
Hi Team, just sharing my thoughts here.

1. The popover element have `absolute` positioning.
<img width="570" alt="image" src="https://github.com/AlaskaAirlines/auro-popover/assets/124668226/1ccf60c4-a535-4b9b-b9c3-f09fc3677662">

2. And it is a child to the `auro-drawer` content div.
<img width="570" alt="image" src="https://github.com/AlaskaAirlines/auro-popover/assets/124668226/efc56dad-8155-41f3-a9d5-fee3d9330277">


Since I believe in `relative-absolute` position theory, then I tried to do that to the `auro-drawer` `content` slot. 
It does work:
<img width="1440" alt="image" src="https://github.com/AlaskaAirlines/auro-popover/assets/124668226/f3e3a40c-2a25-4564-b780-d3e48a78c648">

Although, the popover get cropped, but it does became centered to the trigger element.

**Comment by blackfalcon** [comment]
@fajar-apri-alaska you wanna work up a PR solution to this and a way to test in the repo demo? I did this way back when I validated this issue here https://github.com/AlaskaAirlines/auro-popover/issues/41#issuecomment-1555174013

All that's needed is to add auro-popover to the dialog element temporarily using the CDN version.

**Comment by fajar-apri-alaska** [comment]
> @fajar-apri-alaska you wanna work up a PR solution to this and a way to test in the repo demo? I did this way back when I validated this issue here https://github.com/AlaskaAirlines/auro-popover/issues/41#issuecomment-1555174013
> 
> All that's needed is to add auro-popover to the dialog element temporarily using the CDN version.
> 
> If I am reading your feedback correctly, this is NOT an issue with the popover element, but an update to dialog and drawer?

Yes you are right, @blackfalcon, I think an update is needed to dialog and drawer for above solution. I haven't try any other solution that can be controlled from the `auro-popover` side.
What do you think though? If mine is good enough, I can work on PR for both component right now.

**Comment by braven112** [comment]
Yeah if you have a good idea please go for it.

**Comment by fajar-apri-alaska** [comment]
Hi @braven112 and @blackfalcon,
I have pushed as OSS contributor since I don't have a permission to push for this component yet in [this PR](https://github.com/AlaskaAirlines/auro-interruption/pull/106).

**Comment by jason-capsule42** [comment]
@fajar-apri-alaska I accidently merged this prematurely. There are multiple tests failing with this PR. Can you please resolve the broken tests?

**Comment by fajar-apri-alaska** [comment]
> @fajar-apri-alaska I accidently merged this prematurely. There are multiple tests failing with this PR. Can you please resolve the broken tests?

sure, on it.

**Comment by blackfalcon** [comment]
The attached PR for this work did merge, but it did not deploy. This issue needs to stay open until the release is fixed.
